### PR TITLE
#7 Largest palindrome product 

### PR DESCRIPTION
**Pull request by krishnatimilsina**
Please review the code.

### CODE REVIEW

File: src/com/lftechnology/java/training/krishna/basics/evenfibonacci/EvenFibonacciDemo.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+public class EvenFibonacciDemo {
+	private final static Logger LOGGER = Logger.getLogger(EvenFibonacciDemo.class.getName());
+	
+	/**
+	 * This is the main method which makes use of sumEvenFibonacci() method.
+	 * 
+	 * @param args Unused.
+	 * @return Nothing.
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/krishna/basics/evenfibonacci/EvenFibonacciDemo.java]: it is not required.If it is of void type, you do not have to write @return

File: src/com/lftechnology/java/training/krishna/basics/evenfibonacci/EvenFibonacciDemo.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	 * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+	 */
+	public static void main(String[] args) {
+		EvenFibonacci evenFibonacci = new EvenFibonacci(0, 1);
+		int fib = evenFibonacci.sumEvenFibonacci();
+		LOGGER.info("sum of even fibonacci number : "  + fib);
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/krishna/basics/evenfibonacci/EvenFibonacciDemo.java]: use parameterised logger if you are dealing with the variable.
Have a look at : http://stackoverflow.com/questions/15880629/java-logging-with-multiple-parameters

File: src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	/**
+	 * Check whether or not a given number is palindrome
+	 * 
+	 * @param int n number
+	 * @return boolean
+	 * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+	 */
+	public static boolean isPalindrome(int number) {
+		String s = String.valueOf(number);
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java]: use meaningful variable name

File: src/com/lftechnology/java/training/krishna/basics/palindrome/PalindromeDemo.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+public class PalindromeDemo {
+	private final static Logger LOGGER = Logger.getLogger(PalindromeDemo.class.getName());
+
+	/**
+	 * This is the main method which makes use of findLargestPalindrome() method.
+	 * 
+	 * @param args Unused.
+	 * @return Nothing.
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/krishna/basics/palindrome/PalindromeDemo.java]: same here: not needed

File: src/com/lftechnology/java/training/krishna/basics/palindrome/PalindromeDemo.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	 * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+	 */
+	public static void main(String[] args) {
+		int largestPalindrome = Palindrome.findLargestPalindrome();
+		LOGGER.info("Largest Palindrome : " + largestPalindrome);
```
Comment (Msanjib) [inline src/com/lftechnology/java/training/krishna/basics/palindrome/PalindromeDemo.java]: same here: use parameterised logger

File: src/com/lftechnology/java/training/krishna/basics/smallest_multiple/SmallestMultipleDemo.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package com.lftechnology.java.training.krishna.basics.smallest_multiple;
+
+import java.util.logging.Logger;
+
+public class SmallestMultipleDemo {
```
Comment (utsabneupane) [inline src/com/lftechnology/java/training/krishna/basics/smallest_multiple/SmallestMultipleDemo.java]: class level documentation missing here too.. please avoid repeating such scenario time and again

File: src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java (hunk 3 above)
Comment (krishnatimilsina) [inline src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java]: @Msanjib Do you mean s does not denote String.

Comment (utsabneupane) [inline src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java]: please use proper name for variable

File: src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+		StringBuilder reverseString = new StringBuilder();
+
+		// reverse the string
+		for (int i = s.length() - 1; i > -1; i--) {
+			reverseString.append(s.charAt(i));
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java]: simply use reverseString.reverse() method.

File: src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java (hunk 3 above)
Comment (utsabneupane) [inline src/com/lftechnology/java/training/krishna/basics/palindrome/Palindrome.java]: @krishnatimilsina dai `s` doesnot give meaningful name for variable..

File: src/com/lftechnology/java/training/krishna/basics/smallest_multiple/SmallestMultipleDemo.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+	private final static Logger LOGGER = Logger.getLogger(SmallestMultiple.class.getName());
+
+	public static void main(String[] args) {
+		int number = 2520;
+		SmallestMultiple smallestMultiple = new SmallestMultiple();
+		long positiveNumber = smallestMultiple.smallestPositiveNumber(number);
+		LOGGER.info("smallest positive number : " + positiveNumber);
+
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/krishna/basics/smallest_multiple/SmallestMultipleDemo.java]: unnecessary empty line

File: src/com/lftechnology/java/training/krishna/basics/evenfibonacci/EvenFibonacci.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+ * EvenFibonacci.java
+ * 
+ * The EvenFibonacci program implements an application that simply sum of the even-valued terms
+ * 
+ * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+ * @version 1.0
+ */
+public class EvenFibonacci {
+	private int firstNumber;
+	private int secondNumber;
```
Comment (Satish-Gupta) [inline src/com/lftechnology/java/training/krishna/basics/evenfibonacci/EvenFibonacci.java:13]: Do we really need to make first number and second number a class variable. For fibonacci series the first two numbers are fixed. unless the question also allow to sum fibonacci numbers between a specified range

**Pull request merged**